runtime/shared_common/shrinit.cpp

```diff
@@ -3952,6 +3953,30 @@ j9shr_getCacheSizeBytes(J9JavaVM *vm)
 	return ret;
 }
 
+/**
+ * Check if the shared cache is enabled by default or by using "-Xshareclasses" in the command line. 
+ * 
+ * @param [in] vm Pointer to the VM structure for the JVM
+ *
+ * @return TRUE if shared cache is enabled by default, FALSE otherwise
+ * 
+ */
+J9SharedClassCacheMode
+j9shr_getSharedClassCacheMode(J9JavaVM *vm)
+{
+	J9SharedClassCacheMode ret = JJ9SharedClassCacheUserDefined;
+	/* Only bootstrap classes are shared by default */
+	if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHEBOOTCLASSES) && 
+		J9_ARE_NO_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)) {
+		ret = J9SharedClassCacheBootstrapOnly;
+	}
+	else if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHEBOOTCLASSES) && 
+		J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)) {
+		ret = J9SharedClassCacheUserDefined;
+	}
```

Comment:
This needs only check `J9SHR_RUNTIMEFLAG_ENABLE_CACHEBOOTCLASSES` once:
```
    if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHEBOOTCLASSES)) {
        if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)) {
            ret = J9SharedClassCacheUserDefined;
        } else {
            ret = J9SharedClassCacheBootstrapOnly;
        }
    }
```
Also notice `}` and `else` belong on the same line.